### PR TITLE
Changed the class add command to account for the Cleanup role

### PR DESCRIPTION
**Pull request by Cavesprout**
Ignore the weird commit history, had some issues with github's desktop app being weird

### CODE REVIEW

File: bot/cogs/manage_classes_cog.py [hunk 1]
```diff
@@ -130,7 +130,12 @@ class name as a parameter E.G "cpsc-1010"
         await channel.send(f'Here is your generated class channel {ctx.author.mention}, Good luck!')
 
         #create a class role and mark it as assignable
-        await self.create_role(ctx, class_repr)
+        role = await self.create_role(ctx, class_repr)
+
+        #if cleanup role exists, sync perms
+        if (discord.utils.get(ctx.guild.roles,name="Cleanup")):
```
Comment (Jay-Madden) [inline bot/cogs/manage_classes_cog.py:136]: we need to check if it exists and if  it doesnt then create it and mark it as assignable

File: bot/cogs/manage_classes_cog.py [hunk 2]
```diff
@@ -266,6 +271,15 @@ async def create_role(self, ctx, class_repr):
...
             role = await ctx.guild.create_role(name=class_repr.role, mentionable=False)
         await self.bot.messenger.publish(Events.on_assignable_role_add, role)
+        return role
+
+    async def sync_perms(self, channel, role, cleanup):
+        log.info(f'Syncing channel and role with cleanup')
+        await channel.set_permissions(role, view_channel=True)
+        await channel.set_permissions(cleanup, view_channel=False)
+        role.position=1
+        cleanup.position=0
```
Comment (Jay-Madden) [inline bot/cogs/manage_classes_cog.py:281]: i dont believe setting position here updates it in discord?

Comment (Cavesprout) [reply bot/cogs/manage_classes_cog.py:281]: From what I could tell, though it doesn't update it in the list view, it does indeed give the role permission precedence over cleanup. I could not find a way to update the list view.

Comment (Jay-Madden) [reply bot/cogs/manage_classes_cog.py:281]: https://discordpy.readthedocs.io/en/v0.16.12/api.html#discord.Client.move_role

File: bot/cogs/manage_classes_cog.py (hunk 1 above)
Comment (Cavesprout) [reply bot/cogs/manage_classes_cog.py]: Done

File: bot/cogs/manage_classes_cog.py (hunk 2 above)
Comment (Cavesprout) [reply bot/cogs/manage_classes_cog.py]: Done, move_role() wasn't working but edit() worked

**Review by Jay-Madden — APPROVED** [review]
lgtm

**Pull request merged**